// docusaurus-theme-redoc/dist/theme/Redoc/Redoc.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const merge_1 = __importDefault(require("lodash/merge"));
const redoc_1 = require("redoc");
// @ts-ignore
const useGlobalData_1 = require("@docusaurus/useGlobalData");
const useThemeContext_1 = __importDefault(require("@theme/hooks/useThemeContext"));
require("./styles.css");
/**
 * NOTE: Colors taken from `node_modules/infima/styles/common/dark-mode.css`
 * and related files
 */
const DOCUSAURUS = {
    fontFamily: 'system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
    fontSize: '16px',
    darkGray: '#303846',
    dark: {
        primaryText: '#f5f6f7',
        secondaryText: 'rgba(255, 255, 255, 1)',
        backgroundColor: 'rgb(24, 25, 26)',
    }
};
let LIGHT_THEME_OPTIONS = {
    typography: {
        fontFamily: DOCUSAURUS.fontFamily,
        fontSize: DOCUSAURUS.fontSize,
        headings: {
            fontFamily: DOCUSAURUS.fontFamily,
        },
    },
    sidebar: {
        backgroundColor: '#ffffff',
    },
    rightPanel: {
        backgroundColor: DOCUSAURUS.darkGray,
    }
};
let DARK_THEME_OPTIONS = {
    colors: {
        text: {
            primary: DOCUSAURUS.dark.primaryText,
            secondary: DOCUSAURUS.dark.secondaryText,
        },
        gray: {
            50: '#FAFAFA',
            100: '#F5F5F5',
        },
        border: {
            dark: '#ffffff',
            light: 'rgba(0,0,0, 0.1)',
        },
    },
    schema: {
        nestedBackground: DOCUSAURUS.dark.backgroundColor,
        typeNameColor: DOCUSAURUS.dark.secondaryText,
        typeTitleColor: DOCUSAURUS.dark.secondaryText,
    },
    sidebar: {
        backgroundColor: DOCUSAURUS.dark.backgroundColor,
        textColor: DOCUSAURUS.dark.primaryText,
        arrow: {
            color: DOCUSAURUS.dark.primaryText,
        },
    },
};
function getThemeOptions(baseTheme, isDarkMode) {
    baseTheme = merge_1.default(baseTheme, LIGHT_THEME_OPTIONS);
    if (!isDarkMode)
        return baseTheme;
    return merge_1.default({}, baseTheme, DARK_THEME_OPTIONS);
}
function Redoc(props) {
    const { isDarkTheme } = useThemeContext_1.default();
    const { baseTheme } = useGlobalData_1.usePluginData('docusaurus-theme-redoc');
    const theme = react_1.default.useMemo(() => getThemeOptions(baseTheme, isDarkTheme), [baseTheme, isDarkTheme]);
    const { spec, specUrl } = props;
    return (react_1.default.createElement("div", { className: "redocusaurus" },
        react_1.default.createElement(redoc_1.RedocStandalone, Object.assign({}, (spec ? { spec } : { specUrl }), { options: {
                scrollYOffset: 'nav.navbar',
                hideDownloadButton: true,
                expandSingleSchemaField: true,
                menuToggle: true,
                // @ts-ignore
                suppressWarnings: true,
                theme,
            } }))));
}
exports.default = Redoc;
