// @saucelabs/theme-github-codeblock/build/theme/ReferenceCodeBlock/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseReference = void 0;
const url_1 = require("url");
const react_1 = __importStar(require("react"));
const CodeBlock_1 = __importDefault(require("@theme-init/CodeBlock"));
const initialFetchResultState = {
    code: 'loading...',
    error: null,
    loading: null,
};
const noteStyle = {
    textAlign: 'right',
    fontSize: '.8em'
};
/**
 * parses GitHub reference
 * @param {string} ref url to github file
 */
function parseReference(ref) {
    const fullUrl = ref.slice(ref.indexOf('https'), -1);
    const [url, loc] = fullUrl.split('#');
    /**
     * webpack causes failures when it tries to render this page
     */
    const global = globalThis || {};
    if (!global.URL) {
        // @ts-ignore
        global.URL = url_1.URL;
    }
    const [org, repo, blob, branch, ...pathSeg] = new global.URL(url).pathname.split('/').slice(1);
    const [fromLine, toLine] = loc
        ? loc.split('-').map((lineNr) => parseInt(lineNr.slice(1), 10) - 1)
        : [0, Infinity];
    return {
        url: `https://raw.githubusercontent.com/${org}/${repo}/${branch}/${pathSeg.join('/')}`,
        fromLine,
        toLine,
        title: pathSeg.join('/')
    };
}
exports.parseReference = parseReference;
async function fetchCode({ url, fromLine, toLine }, fetchResultStateDispatcher) {
    let res;
    try {
        res = await fetch(url);
    }
    catch (err) {
        return fetchResultStateDispatcher({ type: 'error', value: err });
    }
    if (res.status !== 200) {
        const error = await res.text();
        return fetchResultStateDispatcher({ type: 'error', value: error });
    }
    const body = (await res.text()).split('\n').slice(fromLine, (toLine || fromLine) + 1);
    const preceedingSpace = body.reduce((prev, line) => {
        if (line.length === 0) {
            return prev;
        }
        const spaces = line.match(/^\s+/);
        if (spaces) {
            return Math.min(prev, spaces[0].length);
        }
        return 0;
    }, Infinity);
    return fetchResultStateDispatcher({
        type: 'loaded',
        value: body.map((line) => line.slice(preceedingSpace)).join('\n')
    });
}
function codeReducer(prevState, { type, value }) {
    switch (type) {
        case 'reset': {
            return initialFetchResultState;
        }
        case 'loading': {
            return { ...prevState, loading: true };
        }
        case 'loaded': {
            return { ...prevState, code: value, loading: false };
        }
        case 'error': {
            return { ...prevState, error: value, loading: false };
        }
        default:
            return prevState;
    }
}
function ReferenceCode(props) {
    const [fetchResultState, fetchResultStateDispatcher] = react_1.useReducer(codeReducer, initialFetchResultState);
    const codeSnippetDetails = parseReference(props.children);
    if (fetchResultState.loading !== false) {
        fetchCode(codeSnippetDetails, fetchResultStateDispatcher);
    }
    const customProps = {
        ...props,
        metastring: ` title="${codeSnippetDetails.title}"`,
        children: initialFetchResultState.code
    };
    return (react_1.default.createElement("div", null,
        react_1.default.createElement(CodeBlock_1.default, Object.assign({}, customProps), fetchResultState.code),
        react_1.default.createElement("div", { style: noteStyle },
            "See full example on ",
            react_1.default.createElement("a", { href: props.children, target: "_blank" }, "GitHub"))));
}
exports.default = ReferenceCode;
